modules/ROOT/pages/dataweave-selecting-types.adoc

```diff
@@ -0,0 +1,204 @@
+= Selecting Types
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Using the dot selector (`.`) over DataWeave xref:dataweave-type-system.adoc#composite-types[types] enables you to declare new types from existing ones. Combined with the xref:dataweave-selectors-reusing-types.adoc[module loader], DataWeave can also load and translate declarations from a custom module file into DataWeave type directives that can be accessed in the same way as types from any other DataWeave module.
+
+== Declare a New Type from an Existing Type
+The following example declares the `NameType` from the `User` type. Using the dot selector in `User.name` to declare the `name` variable a `NameType` enables the `typeOf()`  function to return the primitive type, `"String"`. Notice how the DataWeave variable `name` is declared as a `NameType` value "Seba":
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+type User = {name: String}
+type NameType = User.name
+var name: NameType = "Seba"
+---
+typeOf(name)
+----
+
+.Output: 
+----
+"String"
+----
+
+== Compose a Type from an Existing Type's Field
+The following example uses the dot selector (`.`) to declare a new type `Address` from one key in the complex type `User`:
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+type User = {
+ address : {country: String, city: String, street: String, number: Number},
+ userName : String
+}
+type Address = User.address
+
+var userAddress: Address = {country: "Argentina", city: "Rosario", street: "Calle Falsa", number: 123}
+---
+userAddress
+----
+
+.Output:
+[source,json,linenums]
+----
+{
+ "country": "Argentina",
+ "city": "Rosario",
+ "street": "Calle Falsa",
+ "number": 123
+}
+----
+
+== Select Types from Union Types
+The following example shows how type selection can also be used with `Union` types that consist of `Object` types:
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+type ID = {name: String} | {id: Number}
+var name: ID = {name: "Seba"}
+var id: ID = {id: 38123}
+var nameString: ID.name = "Seba"
+var idNumber: ID.id = 38123
+---
+[name, id, nameString ++ " is of type " ++ typeOf(nameString), idNumber ++ " is of type " ++ typeOf(idNumber)]
```

Comment:
Given the restriction I metioned above I would suggest an example like the following, the current one will yield an error.

```suggestion
%dw 2.0
output application/json
type ID = {id: String, firstName: String, age: Number} | {id: Number, firstName: String, secondName: String}

var nameAge: ID = {firstName: "Seba", id: "123", age: 28}
var fullName: ID = {id: 38123, firstName: "Seba", secondName: "Elizalde"}

var nameString: ID.firstName = "Seba"
var idNumber: ID.id = 38123
---
[nameAge, fullName, nameString ++ " is of type " ++ typeOf(nameString), idNumber ++ " is of type " ++ typeOf(idNumber)]
```